src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -1604,99 +1641,99 @@ void func_80B89280(EnKaizoku* this, PlayState* play) {
         }
     }
 
-    if ((this->unk_2B6 == 0) && (this->unk_2B8 == 0) &&
+    if ((this->iceTimer == 0) && (this->drawDmgEffTimer == 0) &&
         !CHECK_FLAG_ALL(this->picto.actor.flags, ACTOR_FLAG_HOOKSHOT_ATTACHED) &&
         (this->picto.actor.bgCheckFlags & BGCHECKFLAG_GROUND)) {
-        this->unk_2D8 = 0;
-        func_80B85A00(this, play, true);
+        this->dontUpdateSkel = false;
+        EnKaizoku_ReactToPlayer(this, play, true);
 
         if ((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) ||
             (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) {
             Actor_SpawnIceEffects(play, &this->picto.actor, this->bodyPartsPos, KAIZOKU_BODYPART_MAX, 2, 0.7f, 0.4f);
-            this->unk_2B8 = 0;
+            this->drawDmgEffTimer = 0;
             this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
             this->picto.actor.flags |= ACTOR_FLAG_HOOKSHOT_PULLS_PLAYER;
         }
     }
 }
 
-void func_80B893CC(EnKaizoku* this, PlayState* play) {
+void EnKaizoku_SetupDamaged(EnKaizoku* this, PlayState* play) {
     Vec3f sp34;
```

Comment:
This you could just name this `velocity` since it gets directly copied into `this->velocity`